### PR TITLE
Remove MBEDTLS_ENABLE_WEAK_CIPHERSUITES configuration option

### LINKED ISSUE #4416: Remove MBEDTLS_ENABLE_WEAK_CIPHERSUITES configuration option
### Rationale
Remove the configuration option to enable weak ciphersuites -- These suites are dangerous to use and it is highly recommended not to use them and to upgrade to more secure ciphersuites. 

----------------------------------------------------------------
### Work items for 3.0
- Remove all references to MBEDTLS_ENABLE_WEAK_CIPHERSUITES.

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
Fix #4416

### CODE REVIEW

**Review by chris-jones-arm — APPROVED** [review]
LGTM and I cannot find any more references to `MBEDTLS_ENABLE_WEAK_CIPHERSUITES`.

**Review by TRodziewicz — APPROVED** [review]
LGTM

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Missing changelog entry and migration guide

**Comment by ronald-cron-arm** [comment]
Yes I was thinking just the same.

**Comment by ronald-cron-arm** [comment]
I've added the change log and migration guide.

File: ChangeLog.d/remove-enable-weak-ciphersuites.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Removals                                                                        
+   * Removed MBEDTLS_ENABLE_WEAK_CIPHERSUITES configuration option. Fixes #4416.
```
Comment (chris-jones-arm) [inline ChangeLog.d/remove-enable-weak-ciphersuites.txt:2]: Minor:
```suggestion
   * Remove MBEDTLS_ENABLE_WEAK_CIPHERSUITES configuration option. Fixes #4416.
```

**Review by chris-jones-arm — COMMENTED** [review]
The new changelog and migration guide entries look good to me apart from one very minor issue with tense.

File: ChangeLog.d/remove-enable-weak-ciphersuites.txt (hunk 1 above)
Comment (ronald-cron-arm) [reply ChangeLog.d/remove-enable-weak-ciphersuites.txt]: Changed.

**Comment by gilles-peskine-arm** [comment]
CI is not happy.

**Review by chris-jones-arm — APPROVED** [review]
Thanks for the change, LGTM.

File: docs/3.0-migration-guide.d/remove-enable-weak-ciphersuites.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Remove the configuration to enable weak ciphersuites in SSL / TLS
+-----------------------------------------------------------------
+
+This does not affect users who use the default `config.h`, as this option was
+already off by default.
+
+Please switch to any of the modern, recommended ciphersuites (based on
+AES-GCM, AES-CCM or ChachaPoly for example) and if your peer doesn't support
+any, encourage them to upgrade their software.
```
Comment (gilles-peskine-arm) [inline docs/3.0-migration-guide.d/remove-enable-weak-ciphersuites.md:9]: This is accurate for users of ciphersuites with weak crypto such as DES or ARC4. It isn't accurate for users of ciphersuites without encryption. Those are now solely gated by `MBEDTLS_CIPHER_NULL_CIPHER`.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
I reviewed the documentation added after [@TRodziewicz's approval](https://github.com/ARMmbed/mbedtls/pull/4426#pullrequestreview-648717244).

File: docs/3.0-migration-guide.d/remove-enable-weak-ciphersuites.md (hunk 2 above)
Comment (ronald-cron-arm) [reply docs/3.0-migration-guide.d/remove-enable-weak-ciphersuites.md:9]: Thus what do you suggest? I just remove this ?

Comment (gilles-peskine-arm) [reply docs/3.0-migration-guide.d/remove-enable-weak-ciphersuites.md:9]: No, it's correct, but incomplete. Please qualify that sentence with something like “if you were using a weak cipher”, and add a sentence saying that if you were using a ciphersuite without encryption then `MBEDTLS_CIPHER_NULL_CIPHER` alone is enough.

Comment (ronald-cron-arm) [reply docs/3.0-migration-guide.d/remove-enable-weak-ciphersuites.md]: I've amended the last commit.

**Comment by ronald-cron-arm** [comment]
@chris-jones-arm @TRodziewicz @gilles-peskine-arm Please have a look to the last version addressing Gilles' last comment.

**Review by gilles-peskine-arm — APPROVED** [review]

**Review by chris-jones-arm — APPROVED** [review]

**Review by TRodziewicz — COMMENTED** [review]
Looks OK.

**Pull request merged**